docs/components/checkbox/_spec.scss

```diff
@@ -0,0 +1,101 @@
+@import "components/checkbox/config";
+
+.checkbox-spec {
+  hx-checkbox-control {
+    > label {
+      > hx-checkbox {
+        display: none;
+      }
+    }
+
+    // ----- PRISTINE -----
+    > input[type="checkbox"] {
+      // imitate :focus
+      &.mock-focus {
+        @include hxCheckboxControl($pseudo-state: focus);
+      }
+
+      // imitate :hover
+      &.mock-hover {
+        @include hxCheckboxControl($pseudo-state: hover);
+
+        ~ label > hx-checkbox {
+          @include hxCheckboxFacade($pseudo-state: hover);
+        }
+      }
+
+      // imitate :disabled:focus
+      &:disabled.mock-focus {
+        @include hxCheckboxControl(disabled, focus);
+      }
+
+      // imitate :disabled:hover
+      &:disabled.mock-hover {
+        @include hxCheckboxControl(disabled, hover);
+
+        ~ label > hx-checkbox {
+          @include hxCheckboxFacade(disabled, hover);
+        }
+      }
+    }
+
+    // ----- INVALID -----
+    &[hx-dirty] {
+      > input[type="checkbox"]:invalid {
+        // imitate :focus
+        &.mock-focus {
+          @include hxCheckboxControl(invalid, focus);
+        }
+
+        // imitate :hover
+        &.mock-hover {
+          @include hxCheckboxControl(invalid, hover);
+
+          ~ label > hx-checkbox {
+            @include hxCheckboxFacade(invalid, hover);
+          }
+        }
+
+        // default invalid styles
+        ~ label > hx-checkbox {
+          @include hxCheckboxFacade(invalid);
+        }
```

Comment:
Is this block needed? There aren't any pseudo classes being mocked.

Reply:
Block removed.